fontforge/sfd.c

```diff
@@ -3030,9 +3079,9 @@ static int SFDDump(FILE *sfd,SplineFont *sf,EncMap *map,EncMap *normal,
     ff_progress_start_indicator(10,_("Saving..."),_("Saving Spline Font Database"),_("Saving Outlines"),
 	    realcnt,i+1);
     ff_progress_enable_stop(false);
-    double version = 3.1;
-    if( !UndoRedoLimitToSave )
-        version = 3.0;
+    double version = 3.2;
+    if (!WritePNGInSFD) version = 3.1;
+    if (!UndoRedoLimitToSave) version = 3.0;
```

Comment:
Hmm, maybe that will result in saving a bad preference. 

It might be better to make version a static global and test against it instead of `WritePNGInSFD`.

Reply:
No, it won't, since it's impossible for both to be true; it will just at worst correct a bad prefs file if it happens.

Reply:
Isn't `UndoRedoLimitToSave` a preference? Can't it be present as an option and set to false by the user, making the guard evaluate to true?

Reply:
I mean, if you want `UndoRedoLimitToSave`, you _must_ not want `WritePNGInSFD`, as they cannot be mixed. You can't have an SFD 3.0 file with PNGs in it.

Reply:
So if the user wants both, they are wrong, and FontForge is write to correct the error in their preferences. I'm very interested in working on #3800, I just don't think we can get it in on time for August 1, so this solution is temporary. It needs a lot of discussion and @frank-trampe needs time to work on it

Reply:
Can you explain this connection? 

```
        { N_("UndoRedoLimitToSave"), pr_int, &UndoRedoLimitToSave, NULL, NULL, '\0', NULL, 0, N_( "The number of undo and redo operations which will be saved in sfd files.\nIf you set this to 0 undo/redo information is not saved to sfd files.\nIf set to -1 then all available undo/redo information is saved without limit.") },
    { N_("WritePNGInSFD"), pr_bool, &WritePNGInSFD, NULL, NULL, 'B', NULL, 0, N_("If your SFD contains images, write them as PNG; this results in smaller SFDs; but was not supported in FontForge versions compiled before July 2019, so older FontForge versions cannot read them.") },
```
What do these have to do with one another?

Reply:
I think there may be some confusion about how the version numbers are supposed to be used. A given version represents the features that *may* occur. If *X* is added in version 1 and *Y* is added in version 2, choosing not to use feature *X* shouldn't preclude using feature *Y*. 

Presumably what you want to do when `!UndoRedoLimitToSave` is true is bump the version number down *if it's already down to 3.1* and not otherwise.